OCPVE-337: Add an architecture diagram (#341)
openshift/lvm-operator: docs/design/architecture.md

# Logical Volume Manager Storage (LVMS) Architecture

![component diagram](http://www.plantuml.com/plantuml/png/VLDDIyD04BtlhnZgHGyzA8gGWxHDn8jLQBKUf8SbcQR1pSwoEoc8-DzDDi4GkknrtWTltkmk74lBJQs4C3B_aYN2x7ctta9AjI6DcjqCVWJ0Ppk5LuDMCjdzC4XLunWjNAQRx6eF5yiikrxTTE2EL5CZF5bg3AobRet_LJtceO1oIjYovCO5Q6zag8kKacPBIddxSVn21Ux5xtIhiwKcCLthlxfTtZ_U-HXKLhbKq3D6hFESTW7p-KCm3oIGeL7qxQ8A7pNE3TOPruJkay27vFo8ONxuCWca5YMZ0o4caRqgqDCeaBg0scTre0RQgQS94LufrCneg4vr0JdfGrN2eL8OCJaxRJB0ReW0fGTyI3vM6WoHSc77U453gJ5j_5Kdy6uAoLqh4LxagXiyACpP0HyHdDS8iK1TTF_w3m00)

> This table provides a view of all the containers within these components:

Reviewer: ```suggestion
The following table provides a view of the containers in these components:
```

|                      |            | [csi-provisioner](https://github.com/kubernetes-csi/external-provisioner/blob/master/doc/design.md) |
|                      |            | [csi-resizer](https://github.com/kubernetes-csi/external-resizer#csi-resizer)                       |
|                      |            | [liveness-probe](https://github.com/kubernetes-csi/livenessprobe#liveness-probe)                    |
|                      |            | [csi-snapshotter](https://github.com/kubernetes-csi/external-snapshotter#csi-snapshotter)           |
| TopoLVM Node         | DaemonSet  | [topolvm-node](https://github.com/topolvm/topolvm/blob/main/docs/topolvm-node.md)                   |
|                      |            | [lvmd](https://github.com/topolvm/topolvm/blob/main/docs/lvmd.md)                                   |
|                      |            | [csi-registrar](https://github.com/kubernetes-csi/node-driver-registrar#node-driver-registrar)      |
|                      |            | [liveness-probe](https://github.com/kubernetes-csi/livenessprobe#liveness-probe)                    |

> This architecture diagram provides a view of how LVMS components work together to enable dynamic provisioning and management of logical volumes using Logical Volume Manager (LVM) within the OpenShift environment. See also [this page](https://github.com/topolvm/topolvm/blob/main/docs/design.md) for further details on the TopoLVM design. 

Reviewer: ```suggestion
This architecture diagram describes how the LVMS components work together to enable dynamic provisioning and management of logical volumes using Logical Volume Manager (LVM) in OpenShift clusters. See also [this page](https://github.com/topolvm/topolvm/blob/main/docs/design.md) for further details on the TopoLVM design. 
```
